src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.js

```diff
@@ -1,23 +1,56 @@
-import React from "react";
+import React, { useEffect, useRef, useState } from "react";
 import PropTypes from "prop-types";
 import Chip from "../../Chip";
 
 import "./filter-panel-section.scss";
 
-const FilterPanelSection = ({ data }) => (
-  <div className="filter-panel-section">
-    {data?.heading && (
-      <h3 className="filter-panel-section__heading">{data.heading}</h3>
-    )}
-    {data?.chips?.map((chip) => (
-      <Chip
-        lead={chip.lead}
-        value={chip.value}
-        key={`${chip.lead}+${chip.value}`}
-      />
-    ))}
-  </div>
-);
+const FilterPanelSection = ({ data }) => {
+  const { chips, heading } = data;
+  const [overflowCounter, setOverflowCounter] = useState(0);
+  const chipWrapper = useRef(null);
+  const viewportWidth = window.innerWidth;
+
+  // if the offsetTop is more than double height of a single chip, consider it
+  // oveflowing
+  const updateFlowCount = function () {
+    const chips = chipWrapper.current.querySelectorAll(".p-chip");
+    let overflowChips = 0;
+    chips.forEach((chip) => {
+      if (chip.offsetTop > chip.offsetHeight * 2) overflowChips++;
```

Comment:
I'm not a fan of the conditional without curlies, it's just asking for issues later (although granted prettier and linters mitigate that) It's just not really worth the 2 character savings :)